src/pages/api/webhooks/saleor/product-updated.ts

```diff
@@ -0,0 +1,68 @@
+import { NextWebhookApiHandler, SaleorWebhook } from "@saleor/app-sdk/handlers/next";
+import { gql } from "urql";
+import { ProductUpdatedWebhookPayloadFragment } from "../../../../../generated/graphql";
+import { saleorApp } from "../../../../../saleor-app";
+
+// Next.js body parser has to be turned off for us to be able to access the raw request body
+// which is required to validate incoming requests
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+// By using the fragment, we'll be able to get TS types for the payload
+// Since this webhook subscribes for product updates, we have to create fragment on
+// type ProductUpdated
+export const ProductUpdatedWebhookPayload = gql`
+  fragment ProductUpdatedWebhookPayload on ProductUpdated {
+    product {
+      id
+      name
+    }
+  }
+`;
+
+// Subscription query which will be used to construct response for our webhook
+export const ExampleProductUpdatedSubscription = gql`
+  ${ProductUpdatedWebhookPayload}
+  subscription ExampleProductUpdated {
+    event {
+      ...ProductUpdatedWebhookPayload
+    }
+  }
+`;
+
+// The SaleorWebhook class will help us define the webhook and provide handlers for API and manifest
+// ProductUpdatedWebhookPayloadFragment is a fragment object generated by the codegen. We are using it to
+// provide webhook payload types
+export const productUpdatedWebhook = new SaleorWebhook<ProductUpdatedWebhookPayloadFragment>({
+  name: "Example product updated webhook",
+  webhookUrl: `/api/webhooks/saleor/product-updated`,
+  asyncEvent: "PRODUCT_UPDATED",
+  apl: saleorApp.apl,
+  subscriptionQueryAst: ExampleProductUpdatedSubscription,
```

Comment:
suggestion: check what is return type of `gql` and maybe use some other terminology than `ast`